python/pyarrow/array.pyx

```diff
@@ -167,6 +167,12 @@ cdef class Array:
 
         return PyObject_to_object(np_arr)
 
+    def to_pylist(self):
+        """
+        Convert to an list of native Python objects.
+        """
+        return [x.as_py() for x in self]
```

Comment:
At some point we may want to do this marshalling in C to avoid the boxing overhead